wherehows-web/app/components/dataset-compliance.ts

```diff
@@ -130,16 +136,25 @@ const changeSetFieldsRequiringReview = arrayFilter<IComplianceChangeSet>(fieldCh
  */
 const initialStepIndex = -1;
 
-export default class DatasetCompliance extends Component {
+/**
+ * Applies the observer to the editStepIndex to trigger the update edit step task
+ */
+const ObservableDecorator = Component.extend({
+  editStepIndexChanged: observer('editStepIndex', function(this: DatasetCompliance) {
```

Comment:
Why an observer here instead of whatever changes the stepIndex also calling the function you're trying to call here? Seems like making an unnecessary reference.

Reply:
snake eating it's tail, infinite loop. the `updateEditStepTask` is a computed property, that's used in the template, as an aside it's also possibly an async operation, but conceptually what you'll get as a result is:
```javascript
{{#if updateEditStepTaskIsRunning}}
  ...
{{else}}
  {{aComputedPropertyThatChangesTheUpdateEditStepTaskRunningState}}
{{/if}}
```

essentially this results in an error rooted in changing the state twice in render loop.